drivers/gpio/gpio_cmsdk_ahb.c

```diff
@@ -40,17 +41,80 @@ struct gpio_cmsdk_ahb_dev_data {
 	sys_slist_t gpio_cb;
 };
 
+static int gpio_cmsdk_ahb_port_get_raw(struct device *dev, u32_t *value)
+{
+	const struct gpio_cmsdk_ahb_cfg * const cfg = dev->config->config_info;
+
+	*value = cfg->port->data;
+
+	return 0;
+}
+
+static int gpio_cmsdk_ahb_port_set_masked_raw(struct device *dev, u32_t mask,
+					 u32_t value)
+{
+	const struct gpio_cmsdk_ahb_cfg * const cfg = dev->config->config_info;
+
+	cfg->port->dataout = (cfg->port->dataout & ~mask) | (mask & value);
+
+	return 0;
+}
+
+static int gpio_cmsdk_ahb_port_set_bits_raw(struct device *dev, u32_t mask)
+{
+	const struct gpio_cmsdk_ahb_cfg * const cfg = dev->config->config_info;
+
+	cfg->port->dataout |= mask;
+
+	return 0;
+}
+
+static int gpio_cmsdk_ahb_port_clear_bits_raw(struct device *dev, u32_t mask)
+{
+	const struct gpio_cmsdk_ahb_cfg * const cfg = dev->config->config_info;
+
+	cfg->port->dataout &= ~mask;
+
+	return 0;
+}
+
+static int gpio_cmsdk_ahb_port_toggle_bits(struct device *dev, u32_t mask)
+{
+	const struct gpio_cmsdk_ahb_cfg * const cfg = dev->config->config_info;
+
+	cfg->port->dataout ^= mask;
+
+	return 0;
+}
+
 static int cmsdk_ahb_gpio_config(struct device *dev, u32_t mask, int flags)
 {
 	const struct gpio_cmsdk_ahb_cfg * const cfg = dev->config->config_info;
 
+	if (((flags & GPIO_INPUT) == 0) && ((flags & GPIO_OUTPUT) == 0)) {
+		return -ENOTSUP;
+	}
+
+	if ((flags & GPIO_PULL_UP) != 0) {
+		return -ENOTSUP;
+	}
+
+	if ((flags & GPIO_PULL_DOWN) != 0) {
+		return -ENOTSUP;
+	}
```

Comment:
We can write it as
```
	if ((flags & (GPIO_PULL_UP | GPIO_PULL_DOWN)) != 0) {
￼		return -ENOTSUP;
￼	}
```
this should generate smaller code size (unless this is already fixed by the compiler optimization step).

Reply:
Changed.